ecosystem/indexer-grpc/indexer-grpc-table-info/src/table_info_service.rs

```diff
@@ -62,11 +67,30 @@ impl TableInfoService {
                 .await;
             let max_version = self.get_max_batch_version(results).unwrap_or_default();
             let versions_processed = max_version - self.current_version + 1;
+            let context = self.context.clone();
+            let backup_restore_operator = self.backup_restore_operator.clone();
+            let start_version = self.current_version;
+            let indexer_async_v2 = self.indexer_async_v2.clone();
+
+            // Try uploading the rocksdb snapshot by taking a full db checkpoint and save it to gcs if found new epoch
+            // running backup logic in a separate thread to not let it block the main thread to parse table info, since
+            // gcs operation could be slow
+            if let Some(op) = backup_restore_operator {
+                tokio::spawn(async move {
```

Comment:
the problem of this code is that the thread may exit with error and there is no way that this can be recovered